src/createAction.js

```diff
@@ -10,35 +9,30 @@ export default function createAction(type, payloadCreator = identity, metaCreato
     'Expected payloadCreator to be a function, undefined or null'
   );
 
-  const finalPayloadCreator = isNull(payloadCreator)
+  const finalPayloadCreator = payloadCreator === null || payloadCreator === identity
     ? identity
-    : payloadCreator;
+    : (head, ...args) => head instanceof Error ?
+        head : payloadCreator(head, ...args);
+  
+  const hasMeta = isFunction(metaCreator);
 
   const actionCreator = (...args) => {
-    const hasError = args[0] instanceof Error;
-
-    const action = {
-      type
-    };
-
-    const payload = hasError ? args[0] : finalPayloadCreator(...args);
-    if (!isUndefined(payload)) {
+    const payload = finalPayloadCreator(...args);
+    const action = { type, error: payload instanceof Error };
```

Comment:
I don't have a strong preference either way. It doesn't change anything for FSA-compliant applications, and I liked the simplicity of this version. I would have put `meta` and `payload` inline as well, but there was no good way to do that given the limitations of javascript syntax.